src/applications/letters/tests/actions/letters.unit.spec.jsx

```diff
@@ -502,3 +506,95 @@ describe('updateBenefitSummaryRequestOption', () => {
     });
   });
 });
+
+describe('getTsaLetterEligibility', () => {
+  let recordEventStub;
+
+  beforeEach(() => {
+    setup();
+    recordEventStub = sinon.stub(recordEventModule, 'default');
+  });
+
+  afterEach(() => {
+    recordEventStub.restore();
+  });
+
+  it('dispatches LOADING action first', async () => {
+    setFetchJSONResponse(global.fetch.onCall(0), { data: [] });
+    const dispatch = sinon.spy();
+    const thunk = getTsaLetterEligibility();
+    await thunk(dispatch);
+    const action = dispatch.firstCall.args[0];
+    expect(action.type).to.equal(GET_TSA_LETTER_ELIGIBILITY_LOADING);
+  });
+
+  it('dispatches SUCCESS action when fetch succeeds for determining eligibility', async () => {
+    setFetchJSONResponse(global.fetch.onFirstCall(), {
+      data: [
+        {
+          attributes: {
+            documentId: '456',
+          },
+        },
+        {
+          attributes: {
+            documentId: '123',
+            receivedAt: '2025-01-01',
+          },
+        },
+        {
+          attributes: {
+            documentId: '789',
+            receivedAt: '2024-01-01',
+          },
+        },
+      ],
+    });
+    const dispatch = sinon.spy();
+    const thunk = getTsaLetterEligibility();
+    await thunk(dispatch);
+    const successAction = dispatch.secondCall.args[0];
+    expect(successAction.type).to.equal(GET_TSA_LETTER_ELIGIBILITY_SUCCESS);
+    expect(successAction.data.attributes.documentId).to.equal('123');
```

Comment:
successAction.data is not equal to the data seen above in the JSON response. successAction.data is the filtered results of the network response. Here we have 3 letters in the network response, but dispatch only receives an object containing the latest letter.